src/utility.cpp

```diff
@@ -227,6 +227,8 @@ namespace azure {  namespace storage_lite {
             {
                 ret[c] = 1;
             }
+            // Literal + needs to be encoded
+            ret['+'] = 0;
             // Surprisingly, '=' also needs to be encoded because Azure Storage server side is so strict.
             ret['='] = 0;
```

Comment:
Minor: L200 & L201 are not necessary if `/` is added to `subdelimiters`. Also, it is difficult for a reader to understand why there are two additional strings in L196 and L226 instead of them being `subdelimiters`.
Another minor thing to optimize is L205 and L237, at least we can reserve the string's size to avoid some re-allocation right?

Reply:
> L200 & L201 are not necessary if / is added to subdelimiters

sub-delims are defined in RFC 3986. It is a very bad idea to have a different definition in our code base.

Reply:
> at least we can reserve the string's size by the original un-encoded size to avoid some re-allocation right

I don't think so. We cannot forecast how long the encoded string would be before actually encoding it. It's actually trade off between time and space.